### PR TITLE
Fix complex groups context for jms models

### PR DESCRIPTION
**Pull request by 7thcubic**
#1112

### CODE REVIEW

File: ModelDescriber/JMSModelDescriber.php [hunk 1]
```diff
@@ -87,8 +87,10 @@ public function describe(Model $model, Schema $schema)
                     continue;
                 }
 
+                $groups = array_key_exists($name, $model->getGroups()) && is_array($model->getGroups()[$name])? $model->getGroups()[$name] : $model->getGroups();
```
Comment (GuilhemN) [inline ModelDescriber/JMSModelDescriber.php]: There's a space missing after the `is_array` call.

Comment (7thcubic) [reply ModelDescriber/JMSModelDescriber.php]: space??? no its correct it is to make sure that it is an array

**Comment by 7thcubic** [comment]
not sure if it is me but at least for my use case like what is set on the jms controller the group context doesn't seem to be correct. 

because when serializing a child object from a parent object, the context used seems to be the context of the parent object, which means you would not have control of child object context groups like what JMS depicts, meaning if child and parent has the same group name the configuration will bleed over.

File: ModelDescriber/JMSModelDescriber.php (hunk 1 above)
Comment (GuilhemN) [reply ModelDescriber/JMSModelDescriber.php]: I'm talking about a CS issue.

File: ModelDescriber/JMSModelDescriber.php [hunk 2]
```diff
@@ -95,8 +95,14 @@ public function describe(Model $model, Schema $schema)
                     continue;
                 }
 
+                if(empty($model->getGroups()) || !array_key_exists($name, $model->getGroups()) || !is_array($model->getGroups()[$name]))
```
Comment (GuilhemN) [inline ModelDescriber/JMSModelDescriber.php]: The `empty` check is useless

File: ModelDescriber/JMSModelDescriber.php [hunk 3]
```diff
@@ -95,8 +95,14 @@ public function describe(Model $model, Schema $schema)
...
+                    $groups = $model->getGroups();
+                else
```
Comment (GuilhemN) [inline ModelDescriber/JMSModelDescriber.php]: Please use brackets

**Comment by GuilhemN** [comment]
> not sure if it is me but at least for my use case like what is set on the jms controller the group context doesn't seem to be correct. 

Indeed I did not think about this feature when I created this class.

**Comment by GuilhemN** [comment]
Thank you @7thcubic!

**Pull request merged**